src/types/idx_or_name.rs

```diff
@@ -42,11 +43,32 @@ impl FromPyObject<'_> for IdxOrName {
     }
 }
 
-impl ToPyObject for IdxOrName {
```

Comment:
Can't you do
```rs
impl<'py, T> IntoPyObject<'py> for T where T: AsRef<IdxOrName>
```
to avoid this duplication?

Reply:
This is unfortunately not possible,as `IdxOrName` is not generic over `T`, which causes `T` to be considered a foreign type :confused: 

```
 impl<'py, T: AsRef<IdxOrName>> IntoPyObject<'py> for T {
           ^ type parameter `T` must be used as the type parameter for some local type

 note: implementing a foreign trait is only possible if at least one of the types for which it is implemented is local
 note: only traits defined in the current crate can be implemented for a type parameter
```

Reply:
Ah fk yes. The orphan rule striking again. LGTM then